src/actions/reactions/bookmark.js

```diff
@@ -0,0 +1,49 @@
+import toastr from 'toastr';
+import axios from '../../utils/axiosInstance';
+
+import {
+  BOOKMARK,
+  BOOKMARK_ERROR,
+  BOOKMARK_ON_ALL,
+  UNBOOKMARK,
+  UNBOOKMARK_ERROR,
+  UNBOOKMARK_ON_ALL,
+} from '../types';
+
+export const bookmarkArticle = (slug, articlePage, id) => async (dispatch) => {
+  try {
+    const { data } = await axios.post(`/articles/${slug}/bookmarks`);
+
+    if (articlePage === 'allArticles') {
+      dispatch({ type: BOOKMARK_ON_ALL, payload: { data, id } });
+      return toastr.success('Bookmarked article successfully');
```

Comment:
I think we could do a better job with the success message. What do you think about 'Article bookmarked successfully?'

Reply:
@frostyblok, My idea was to show the user the action that happened first, in the statement. Your suggestion also makes sense though

Reply:
Good!